pandas/core/indexes/range.py

```diff
@@ -348,6 +348,34 @@ def get_loc(self, key, method=None, tolerance=None):
                 raise KeyError(key)
         return super().get_loc(key, method=method, tolerance=tolerance)
 
+    @Appender(_index_shared_docs['get_indexer'])
+    def get_indexer(self, target, method=None, limit=None, tolerance=None):
+        if method is None and tolerance is None and is_list_like(target):
+            if self.step > 0:
+                start, stop, step = self.start, self.stop, self.step
+            else:
+                # Work on reversed range for simplicity:
+                start, stop, step = (self.stop - self.step,
+                                     self.start + 1,
+                                     - self.step)
+
+            target_array = np.asarray(target)
+            if target_array.dtype != int:
+                # checks/conversions/roundings are delegated to general method
+                return super().get_indexer(target, method=method,
```

Comment:
i would this check near the top as well

Reply:
I need target to already be an array (or to duplicate dtype checking logics)